src/Bag.jl

```diff
@@ -0,0 +1,55 @@
+# This file is copied from YaoExtensions for convenience,
+# will be considerd to move to YaoBlocks!
+export AbstractBag
+"""
+    AbstractBag{BT, N}<:TagBlock{BT, N}
+
+Abstract `Bag` is a wrapper of a block that conserves all properties.
+Including `mat`, `apply!`, `ishermitian`, `isreflexive`, `isunitary`,
+`occupied_locs`, `apply_back!` and `mat_back!`.
+"""
+abstract type AbstractBag{BT, N}<:TagBlock{BT, N} end
+
+YaoBlocks.mat(::Type{T}, bag::AbstractBag{N}) where {T,N} = mat(T, content(bag))
+YaoBlocks.apply!(reg::AbstractRegister, bag::AbstractBag) = apply!(reg, content(bag))
+YaoBlocks.ishermitian(bag::AbstractBag) = ishermitian(content(bag))
+YaoBlocks.isreflexive(bag::AbstractBag) = isreflexive(content(bag))
+YaoBlocks.isunitary(bag::AbstractBag) = isunitary(content(bag))
+YaoBlocks.occupied_locs(bag::AbstractBag) = occupied_locs(content(bag))
+
+function Yao.AD.apply_back!(state, b::AbstractBag, collector)
```

Comment:
`YaoBlocks.AD` this is a small component package,  it shouldn't depend on Yao (the meta package).

Reply:
Can you fix this so the CI passes? @GiggleLiu